Source/Scene/Expression.js

```diff
@@ -13,14 +17,128 @@ define([
     function Expression(styleEngine, expression) {
         this._styleEngine = styleEngine;
 
-        console.log(jsep(expression));
+        this._ast = jsep(expression);
+        this._runtimeAST = undefined;
+
+        console.log(this._ast);
+        this._runtimeAST = createRuntimeAst(this._ast);
+        console.log(this._runtimeAST);
+    }
+
+    function createRuntimeAst(ast) {
+        var node = ast;
+
+        // evaluate this node if we can
+        if (ast.type === 'Literal') {
+            if (typeof(ast.value) === 'string') {
+                var c = Color.fromCssColorString(ast.value);
+                if (defined(c)) {
+                    node.value = c;
+                }
+            }
+        } else if(ast.type === 'UnaryExpression') {
+            var operand = createRuntimeAst(ast.argument);
+            if (operand.type === 'Literal') {
+                node = defineLiteralNode(evaluateUnary(ast.operator, operand.value));
+            }
+        } else if (ast.type === 'BinaryExpression') {
+            var left = createRuntimeAst(ast.left);
+            var right = createRuntimeAst(ast.right);
+            if (left.type === 'Literal' && right.type === 'Literal') {
+                node = defineLiteralNode(evaluateBinary(left.value, ast.operator, right.value));
+            }
+        } else if (ast.type === 'LogicalExpression') {
+            var l = createRuntimeAst(ast.left);
+            var r = createRuntimeAst(ast.right);
+            if (l.type === 'Literal' && r.type === 'Literal') {
+                node = defineLiteralNode(evaluateLogical(l.value, ast.operator, r.value));
+            }
+        } else if (ast.type === 'ConditionalExpression') {
+            var test = createRuntimeAst(ast.test);
+            if (test.type === 'Literal') {
+                node = createRuntimeAst(evaluateConditional(test.value, ast.consequent, ast.alternate));
+            }
+
+        }
+
+        return node;
     }
 
     defineProperties(Expression.prototype, {
     });
 
+    function defineLiteralNode(value) {
```

Comment:
Functions like this should start with `create` instead of `define`.  However, for performance, we do not want to use an object literal here; instead we want to create a constructor function to define a new type so a hidden class is created.

For now, the new type can be local to this file, but as we progress, we'll move it (and the other new AST types) to new files and add reference documentation since these will be exposed in the Cesium API.